pcm_wav.cpp

```diff
@@ -45,13 +51,13 @@ PcmOutputWav::PcmOutputWav(const char *fn)
     header.nBitsPerSample = 16;
     strncpy(header.tagdata, "data", 4);
     header.datasize = 0;
-
-    f = fopen(fn, "wb");
+    Filename = fn;
```

Comment:
Filename is a pointer, so this may be undefined behavior (because you don't know what it points to, it could be a local variable). Nothing uses the Filename member anyway, so remove it and revert this section of the code back to the way it was.